appcelerator/titanium
# feat(sdk): pull directly from github releases for GA builds

Note:

I just noticed that `ti setup check` calls `http://api.appcelerator.com/p/v1/release-list'` so putting in draft for now while I work to replace that call with a GitHub call.

JIRA: https://jira.axway.com/browse/TIMOB-21152

* Query for releases on the appcelerator/titanium_mobile repository instead of using the mobilesdk endpoint
  * Adds new `sdk.github.owner` and `sdk.github.repo` config options just to futureproof incase the release endpoints do not redirect
  * Uses @octokit/rest as it handles pagination and auth for us easier than calling the APIs ourselves
  * Supports a `GITHUB_TOKEN` environment variable to allow people to override the [default rate limit](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting) of 60 requests an hour per IP
* Adds a warning when running `ti sdk install --branch`, and adds handling for errors on that call

To test:

* Run `ti sdk install latest` - It should install an SDK as normal
* Run `ti sdk --releases` until you get an error about hitting the GitHub API limit. Then set a `GITHUB_TOKEN` environment variable with an [auth token](https://github.com/settings/tokens), rerun `ti sdk --releases` and it should work.
* Run `ti sdk install --branch master` it should log a warning
* Add `127.0.0.1 f4df0833a1352ef193633d85356e6e80ffcdcdb7.cloudapp-enterprise.appcelerator.com` to your `/etc/hosts` file, and run `ti sdk install --branch master` it should error with a message about branch builds no longer being available
